addons/website_sale/models/product.py

```diff
@@ -200,7 +200,7 @@ def _website_price(self):
         ret = self.env.user.has_group('sale.group_show_price_subtotal') and 'total_excluded' or 'total_included'
 
         for p, p2 in zip(self, self2):
-            taxes = partner.property_account_position_id.map_tax(p.taxes_id).filtered(lambda x: x.company_id == company_id)
+            taxes = partner.property_account_position_id.map_tax(p.sudo().taxes_id.filtered(lambda x: x.company_id == company_id))
```

Comment:
@pedrobaeza The problem I'm finding here is not with multi company buy with permissions. In the case of a default `public user` session: `product` is loaded with `public user` `uid` and can't run methods from taxes object due to permissions. Instead, partner is loaded with `admin` `uid`, so when `filtered()`  is piped on it permissions aren't a problem.

As you pointed before this issue roots are deeper than what this patch solves so maybe the solution must be refocused.